app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -190,6 +193,11 @@ def product_variant(product)
     end
   end
 
+  def remove_product_from_order_cycles
+    variant_ids_csv = @product.variants.map(&:id).join(', ')
+    ActiveRecord::Base.connection.execute("DELETE FROM exchange_variants WHERE variant_id IN (#{variant_ids_csv})")
+  end
```

Comment:
if we agree this is the way to go I can move this method a a service ExchangeVariantDeleter